src/util/fuzz_utils.rs

```diff
@@ -0,0 +1,1001 @@
+// Rust Monero Library
+// Written in 2019-2023 by
+//   Monero Rust Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+
+use crate::blockdata::transaction::{ExtraField, KeyImage, RawExtraField, SubField, TxOutTarget};
+use crate::consensus::encode::Encodable;
+use crate::consensus::{deserialize, serialize};
+use crate::cryptonote::hash::Hashable;
+use crate::util::key::H;
+use crate::util::ringct::{CtKey, EcdhInfo, Key, RctSig, RctSigBase, RctSigPrunable, RctType};
+use crate::{
+    Amount, Block, BlockHeader, Hash, PrivateKey, PublicKey, Transaction, TransactionPrefix, TxIn,
+    TxOut, VarInt, ViewPair,
+};
+use hex::{FromHex, ToHex};
+use std::io;
+use std::str::FromStr;
+
+/// Fuzz for block deserialization, called from the fuzz target
+pub fn fuzz_block_deserialize(fuzz_data: &[u8]) -> bool {
+    let fuzz_bytes = fuzz_data.to_vec();
+
+    // Block
+    if let Ok(val) = deserialize::<Block>(&fuzz_bytes[..]) {
+        println!("here 1");
+        println!("header {:?}", serialize(&val.header));
+        println!("transaction_prefix {:?}", serialize(&val.miner_tx.prefix));
+        assert_eq!(fuzz_bytes, serialize(&val), "\nfuzz_data: {:?}", fuzz_data);
+    }
+
+    true
+}
+
+/// Fuzz for block header deserialization, called from the fuzz target
+pub fn fuzz_block_header_deserialize(fuzz_data: &[u8]) -> bool {
+    let fuzz_bytes = fuzz_data.to_vec();
+
+    // BlockHeader
+    if let Ok(val) = deserialize::<BlockHeader>(&fuzz_bytes[..]) {
+        println!("here 2");
```

Comment:
I don't think these `println!("here X");` are needed

Reply:
Yeah, my bad, used them to improve the test coverage